lib/compress/zstd_ldm.h

```diff
@@ -48,15 +47,17 @@ size_t ZSTD_ldm_generateSequences(
  * block compressor. The literals section of every sequence is passed to the
  * secondary block compressor, and those sequences are interspersed with the
  * predefined sequences. Returns the length of the last literals.
- * `nbSeq` is the number of sequences available in `sequences`.
```

Comment:
Just trying to understand the API at this stage.

Before calling `ZSTD_ldm_blockCompress()`: 
`*nbSeq` == nb of sequences in `sequences`.
So, it's basically the size of a table of `rawSeq`.

After calling `ZSTD_ldm_blockCompress()`: 
`*nbSeq` == nb of sequences consumed within `sequences`.
Hence, the index of the beginning of the remaining.

It works.
But I'm slightly annoyed by the change of nature of `*nbSeq`,
which starts by representing an "end of table", and then becomes a "beginning of table".
It's usable, it just pushes extra care into user territory.

The situation feels somewhat similar to streaming interface, which adopted `ZSTD_{in,out}Buffer*` format for its API. In this format, beginning and end of table are preserved after each function call, only `pos`, the starting position to read or write into the table, is updated.
This sounds like a little detail, but was pretty effective to make the API easier to read and use.

Does that sound applicable here ? Would it make the interface easier to use ?

Reply:
Yeah thats make sense, I'll update it.